File: payroll-saas-frontend/src/api/payroll.ts
import axios from "axios";

const API_URL = "http://localhost:8000/payroll";

interface PayrollInput {
  base_salary: number;
  allowances: number;
}

export const simulatePayroll = async (input: PayrollInput) => {
  try {
    const res = await axios.post(`${API_URL}/simulate/`, input);
    return res.data;
  } catch (err) {
    console.error("Error simulating payroll:", err);
    throw err;
  }
};

File: payroll_saas_backend/app/api/payroll.py
from fastapi import APIRouter
from app.models.schemas import PayrollInput
from app.services.payroll_service import run_simulation

router = APIRouter(prefix="/payroll", tags=["Payroll Simulator"])

@router.post("/simulate")
def simulate_payroll(payload: PayrollInput):
    result = run_simulation(payload.base_salary, payload.allowances)
    return {"status": "success", "data": result}
